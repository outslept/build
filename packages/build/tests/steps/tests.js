import { platform } from 'process'

import { Fixture, normalizeOutput } from '@netlify/testing'
import { test, expect } from 'vitest'

if (platform !== 'win32') {
  test('build.command uses Bash', async () => {
    const output = await new Fixture('./fixtures/bash').runWithBuild()
    expect(normalizeOutput(output)).toMatchSnapshot()
  })

  test('build.command can execute shell commands', async () => {
    const output = await new Fixture('./fixtures/shell').runWithBuild()
    expect(normalizeOutput(output)).toMatchSnapshot()
  })
}

test('build.command can execute global binaries', async () => {
  const output = await new Fixture('./fixtures/global_bin').runWithBuild()
  expect(normalizeOutput(output)).toMatchSnapshot()
})

test('build.command can execute local binaries', async () => {
  const output = await new Fixture('./fixtures/local_bin').runWithBuild()
  expect(normalizeOutput(output)).toMatchSnapshot()
})

test('build.command use correct PWD', async () => {
  const output = await new Fixture('./fixtures/pwd').runWithBuild()
  expect(normalizeOutput(output)).toMatchSnapshot()
})

test('build.command from UI settings', async () => {
  const output = await new Fixture('./fixtures/none')
    .withFlags({ defaultConfig: { build: { command: 'node --version' } } })
    .runWithBuild()
  expect(normalizeOutput(output)).toMatchSnapshot()
})

test('Invalid package.json does not make build fail', async () => {
  const output = await new Fixture('./fixtures/invalid_package_json').runWithBuild()
  expect(normalizeOutput(output)).toMatchSnapshot()
})
