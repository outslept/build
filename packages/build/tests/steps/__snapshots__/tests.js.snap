// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Invalid package.json does not make build fail 1`] = `
"
Netlify Build                                                 
────────────────────────────────────────────────────────────────

> Version
  @netlify/build 1.0.0

> Flags
  debug: true
  repositoryRoot: packages/build/tests/steps/fixtures/invalid_package_json
  testOpts:
    pluginsListUrl: test
    silentLingeringProcesses: true

> Current directory
  packages/build/tests/steps/fixtures/invalid_package_json

> Config file
  No config file was defined: using default values.

> Resolved config
  build:
    publish: packages/build/tests/steps/fixtures/invalid_package_json
    publishOrigin: default

> Context
  production

Netlify Build Complete                                        
────────────────────────────────────────────────────────────────

(Netlify Build completed in 1ms)
Build step duration: Netlify Build completed in 1ms"
`;

exports[`build.command can execute global binaries 1`] = `
"
Netlify Build                                                 
────────────────────────────────────────────────────────────────

> Version
  @netlify/build 1.0.0

> Flags
  debug: true
  repositoryRoot: packages/build/tests/steps/fixtures/global_bin
  testOpts:
    pluginsListUrl: test
    silentLingeringProcesses: true

> Current directory
  packages/build/tests/steps/fixtures/global_bin

> Config file
  packages/build/tests/steps/fixtures/global_bin/netlify.toml

> Resolved config
  build:
    command: node --version
    commandOrigin: config
    publish: packages/build/tests/steps/fixtures/global_bin
    publishOrigin: default

> Context
  production

build.command from netlify.toml                               
────────────────────────────────────────────────────────────────

$ node --version
v1.0.0

(build.command completed in 1ms)
Build step duration: build.command completed in 1ms

Netlify Build Complete                                        
────────────────────────────────────────────────────────────────

(Netlify Build completed in 1ms)
Build step duration: Netlify Build completed in 1ms"
`;

exports[`build.command can execute local binaries 1`] = `
"
Netlify Build                                                 
────────────────────────────────────────────────────────────────

> Version
  @netlify/build 1.0.0

> Flags
  debug: true
  repositoryRoot: packages/build/tests/steps/fixtures/local_bin
  testOpts:
    pluginsListUrl: test
    silentLingeringProcesses: true

> Current directory
  packages/build/tests/steps/fixtures/local_bin

> Config file
  packages/build/tests/steps/fixtures/local_bin/netlify.toml

> Resolved config
  build:
    command: atob dGVzdA==
    commandOrigin: config
    publish: packages/build/tests/steps/fixtures/local_bin
    publishOrigin: default

> Context
  production

build.command from netlify.toml                               
────────────────────────────────────────────────────────────────

$ atob dGVzdA==
test

(build.command completed in 1ms)
Build step duration: build.command completed in 1ms

Netlify Build Complete                                        
────────────────────────────────────────────────────────────────

(Netlify Build completed in 1ms)
Build step duration: Netlify Build completed in 1ms"
`;

exports[`build.command from UI settings 1`] = `
"
Netlify Build                                                 
────────────────────────────────────────────────────────────────

> Version
  @netlify/build 1.0.0

> Flags
  debug: true
  repositoryRoot: packages/build/tests/steps/fixtures/none
  testOpts:
    pluginsListUrl: test
    silentLingeringProcesses: true

> Current directory
  packages/build/tests/steps/fixtures/none

> Config file
  No config file was defined: using default values.

> Resolved config
  build:
    command: node --version
    commandOrigin: ui
    publish: packages/build/tests/steps/fixtures/none
    publishOrigin: default

> Context
  production

Build command from Netlify app                                
────────────────────────────────────────────────────────────────

$ node --version
v1.0.0

(build.command completed in 1ms)
Build step duration: build.command completed in 1ms

Netlify Build Complete                                        
────────────────────────────────────────────────────────────────

(Netlify Build completed in 1ms)
Build step duration: Netlify Build completed in 1ms"
`;

exports[`build.command use correct PWD 1`] = `
"
Netlify Build                                                 
────────────────────────────────────────────────────────────────

> Version
  @netlify/build 1.0.0

> Flags
  debug: true
  repositoryRoot: packages/build/tests/steps/fixtures/pwd
  testOpts:
    pluginsListUrl: test
    silentLingeringProcesses: true

> Current directory
  packages/build/tests/steps/fixtures/pwd

> Config file
  packages/build/tests/steps/fixtures/pwd/netlify.toml

> Resolved config
  build:
    command: node file.js
    commandOrigin: config
    publish: packages/build/tests/steps/fixtures/pwd
    publishOrigin: default

> Context
  production

build.command from netlify.toml                               
────────────────────────────────────────────────────────────────

$ node file.js
true

(build.command completed in 1ms)
Build step duration: build.command completed in 1ms

Netlify Build Complete                                        
────────────────────────────────────────────────────────────────

(Netlify Build completed in 1ms)
Build step duration: Netlify Build completed in 1ms"
`;
